contracts/ExitFormat.sol

```diff
@@ -137,8 +143,17 @@ library ExitFormat {
                 (bool success, ) = destination.call{value: amount}(""); //solhint-disable-line avoid-low-level-calls
                 require(success, "Could not transfer ETH");
             } else {
-                // TODO support other token types via the singleAssetExit.metadata field
-                ERC20Interface(asset).transfer(destination, amount);
+                // Inspect the metadata to see what kind of token to transfer
+                bytes memory _metadata_ = singleAssetExit.metadata;
+                // Empty metadata or 0 => ERC20
+                if( _metadata_.length == 0 || _metadata_[31] == bytes1(uint8(ExitMetadataType.ERC20)) ) {
```

Comment:
This part I am not happy with. The encoding uses the unpacked form so the type byte takes up a full 32 bytes. Hence checking the `31` byte to identify the type.

Someone with more Solidity and ABI encoding knowledge can hopefully suggest a better and more efficient way to do this check